### PR TITLE
Add camel-google-pubsub wrapper

### PR DESCRIPTION
**Pull request by stataru8**
camel-google-pubsub wrapper

### CODE REVIEW

File: features/src/main/feature/camel-features.xml [hunk 1]
```diff
@@ -944,48 +944,22 @@
...
-<!--        <bundle dependency='true'>mvn:org.bouncycastle/bcprov-jdk18on/${bouncycastle-version}</bundle>-->
-<!--        <bundle dependency='true'>mvn:org.bouncycastle/bcutil-jdk18on/${bouncycastle-version}</bundle>-->
-<!--        <bundle>mvn:org.apache.camel/camel-google-pubsub/${project.version}</bundle>-->
-<!--    </feature>-->
+    <feature name='camel-google-pubsub' version='${project.version}' start-level='50'>
+        <feature version='${project.version}'>camel-core</feature>
+        <bundle dependency='true'>wrap:mvn:com.google.cloud/google-cloud-pubsub/1.126.2</bundle>
+        <bundle dependency='true'>wrap:mvn:com.google.api/api-common/2.24.0</bundle>
+        <bundle dependency='true'>mvn:com.google.protobuf/protobuf-java/3.25.2</bundle>
+        <bundle dependency='true'>mvn:com.google.guava/guava/33.0.0-jre</bundle>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]: The expected version of guava is `32.1.3-jre` according to the bom of google-cloud-pubsub v 1.126.2 as you can see https://central.sonatype.com/artifact/com.google.cloud/google-cloud-pubsub/1.126.2

File: features/src/main/feature/camel-features.xml [hunk 2]
```diff
@@ -944,48 +944,22 @@
...
+        <bundle dependency='true'>mvn:com.google.guava/failureaccess/1.0.1</bundle>
+        <bundle dependency='true'>wrap:mvn:com.google.api/gax/2.41.0</bundle>
+        <bundle dependency='true'>wrap:mvn:com.google.api/gax-grpc/2.41.0</bundle>
+        <bundle dependency='true'>wrap:mvn:com.google.auth/google-auth-library-oauth2-http/1.22.0</bundle>
+        <bundle dependency='true'>wrap:mvn:com.google.auth/google-auth-library-credentials/1.22.0</bundle>
+        <bundle dependency='true'>wrap:mvn:com.google.api.grpc/proto-google-cloud-pubsub-v1/1.22.0</bundle>
```
Comment (essobedo) [inline features/src/main/feature/camel-features.xml]: The expected version is `1.108.2` still according to the previous bom file

File: features/src/main/feature/camel-features.xml (hunk 1 above)
Comment (stataru8) [reply features/src/main/feature/camel-features.xml]: Yes, the plugin is also telling me the same, but I'm unsure if we should use the same version as in camel: https://github.com/apache/camel/blob/camel-4.4.1/parent/pom.xml#L193

Comment (stataru8) [reply features/src/main/feature/camel-features.xml]: also, on my side
```
[INFO] org.apache.camel:camel-google-pubsub:jar:4.4.1
[INFO] +- org.apache.camel:camel-support:jar:4.4.1:compile
[INFO] |  +- org.apache.camel:camel-api:jar:4.4.1:compile
[INFO] |  |  \- jakarta.xml.bind:jakarta.xml.bind-api:jar:4.0.1:compile
[INFO] |  |     \- jakarta.activation:jakarta.activation-api:jar:2.1.2:compile
[INFO] |  +- org.apache.camel:camel-management-api:jar:4.4.1:compile
[INFO] |  +- org.apache.camel:camel-util:jar:4.4.1:compile
[INFO] |  +- org.apache.camel:camel-util-json:jar:4.4.1:compile
[INFO] |  +- org.apache.camel:camel-xml-jaxp-util:jar:4.4.1:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:2.0.12:compile
[INFO] +- com.google.cloud:google-cloud-pubsub:jar:1.126.2:compile
[INFO] |  +- io.grpc:grpc-api:jar:1.61.0:compile
[INFO] |  +- io.grpc:grpc-stub:jar:1.61.0:runtime
[INFO] |  +- io.grpc:grpc-protobuf:jar:1.61.0:compile
[INFO] |  +- io.grpc:grpc-protobuf-lite:jar:1.61.0:runtime
[INFO] |  +- com.google.api:api-common:jar:2.24.0:compile
[INFO] |  +- com.google.protobuf:protobuf-java:jar:3.25.2:compile
[INFO] |  +- com.google.api.grpc:proto-google-common-protos:jar:2.32.0:compile
[INFO] |  +- com.google.auth:google-auth-library-oauth2-http:jar:1.22.0:compile
[INFO] |  +- com.google.auth:google-auth-library-credentials:jar:1.22.0:compile
[INFO] |  +- com.google.http-client:google-http-client-gson:jar:1.43.3:compile
[INFO] |  +- com.google.api.grpc:proto-google-cloud-pubsub-v1:jar:1.108.2:compile
[INFO] |  +- com.google.api.grpc:proto-google-iam-v1:jar:1.27.0:compile
[INFO] |  +- com.google.guava:guava:jar:33.0.0-jre:compile
[INFO] |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
```

File: features/src/main/feature/camel-features.xml (hunk 2 above)
Comment (stataru8) [reply features/src/main/feature/camel-features.xml]: Fix applied in last commit

**Review by essobedo — APPROVED** [review]
LGTM

**Pull request merged**